jwst/photom/photom.py

```diff
@@ -468,7 +470,32 @@ def calc_miri(self, ftab):
                 row = find_row(ftab.phot_table, fields_to_match)
                 if row is None:
                     return
-            self.photom_io(ftab.phot_table[row])
+                self.photom_io(ftab.phot_table[row])
```

Comment:
Is the indent level of this line correct? With it indented the way it is, it'll be executed as part of the `if` block that begins in line 466, which is only executed if the first call to `find_row` (in line 465) fails to find a matching row and returns None. But what if it succeeds in finding a match? It appears that `self.photom_io()` won't get called at all in that case.

Reply:
Yes. That part is present in the master branch I pulled the code from. My understanding that if the photom reference file is old and does not have the newest subarray keywords, the code searches for the needed row using a different keyword ("GENERIC"). If not even that keyword is present, then the code basically skips the photom step (returns None).

Reply:
But the existing code on master https://github.com/spacetelescope/jwst/blob/master/jwst/photom/photom.py#L471 shows that line at the same indent level as line 464, but this PR branch has it indented an extra 4 spaces, as shown in the diff above. So it's now part of the 2nd if-block, whereas before it was not part of either of them and hence would still get executed even if the 2nd if-block got skipped. Now, if the 2nd if-block gets skipped, that line won't get executed.

Reply:
OK. The reason is the following. If the first call at line 465 works, then it means that the reference file has a subarray keyword other than GENERIC, which means the photom ref file is relatively new. In this case, we need to check if there is the new timecoeff table. If there is not, the code executes as before.

The reason of the extra indent is that if the first call at line 465 fails, it automatically means that the photom file does not have the timecoeff table. It is thus unnecessary to continue even if a match with subarray=GENERIC is found. This way the code skips one check.

Does it make sense?

Reply:
Yes, I understand now. So the code layout in the PR is OK. Thanks for clarification.